virtual_rainforest/main.py

```diff
@@ -0,0 +1,125 @@
+"""Defines the function used to run a full simulation of the model.
+
+As well as setting up the function to run the overall virtual rainforest simulation,
+this script also defines the command line entry points for the model.
+"""
+
+from copy import deepcopy
+from typing import Any, Type, Union
+
+from virtual_rainforest.core.config import validate_config
+from virtual_rainforest.core.logger import LOGGER, log_and_raise
+from virtual_rainforest.core.model import (
+    MODEL_REGISTRY,
+    BaseModel,
+    FailedModel,
+    InitialisationError,
+)
+
+
+def select_models(model_list: list[str]) -> list[Type[BaseModel]]:
+    """Select the models to be run for a specific virtual rainforest simulation.
+
+    This function looks for models from a list of models, if these models can all be
+    found in the registry then they are returned. Otherwise an error is logged, which
+    should be handled appropriately downstream.
+
+    Args:
+        model_list: A list of models to select
+
+    Raises:
+        InitialisationError: If one or more models cannot be found in the registry
+    """
+
+    # Remove "core" from model list as it is not a model
+    if "core" in model_list:
+        model_list.remove("core")
+
+    LOGGER.info("Attempting to configure the following models: %s" % model_list)
+
+    # Make list of missing models, and return an error if necessary
+    miss_model = [model for model in model_list if model not in MODEL_REGISTRY.keys()]
+    if miss_model != []:
```

Comment:
Not sure if it is faster or not, but I usually prefer:

```suggestion
    if len(miss_model):
```